packages/fuel-gauge/src/abi/vitest.matcher.ts

```diff
@@ -1,20 +1,38 @@
 import { bn } from 'fuels';
-import type { BNInput } from 'fuels';
+import type { BNInput, BN } from 'fuels';
 
-export const toEqualBn = (_received: BNInput, _argument: BNInput) => {
-  const received = bn(_received);
-  const argument = bn(_argument);
+interface Matchers<R = BN> {
+  toEqualBn: (expected: BNInput) => R;
+}
 
-  const pass = received.eq(argument);
-
-  if (pass) {
-    return {
-      message: () => `Expected ${received.toString()} not to equal ${argument.toString()}`,
-      pass: true,
-    };
+declare module 'vitest' {
+  interface Assertion extends Matchers {}
+  interface AsymmetricMatchersContaining extends Matchers {}
+  interface ExpectStatic {
+    toEqualBn(expected: BNInput): BN;
   }
-  return {
-    message: () => `expected ${received.toString()} to equal ${argument.toString()}`,
-    pass: false,
-  };
+}
```

Comment:
- Replace the contents of `packages/fuel-gauge/test/global.d.ts` with this.
- Add the [following](https://github.com/FuelLabs/fuels-ts/pull/3438/commits/8ed2d14d5f57581746b260b4d83364af3a01c479) to the `tsconfig.test.json`
`git cherry-pick 8ed2d14d5f57581746b260b4d83364af3a01c479`.

I believe this should resolve the linting issue.